### PR TITLE
feature: support auto-cloning of source control backed services

### LINKED ISSUE #7: feature: support auto-cloning of source control backed services
We like that Tarantino can clone your source control repositories on the fly. We also like that Tarantino doesn't try and clone repositories for services that aren't source control backed (like mongo, redis, etc.). 

We have a few responsibilities that need to be filled one way or another, implicitly or explicitly:

  + know which services are SCM backed (and which are not)
  + know the URL to the source control repository for each service
  + know the type of source control repository (git, svn, mercurial)
 
Our original Tarantino model assumed a few things about source control:

  1. all repositories use Git
  2. all repositories belong to one GitHub organization
  3. all repository names precisely match `docker-compose` service names

We need to decide how to add this functionality. We need to be cognizant that changing how we fulfill the set of responsibilities noted above will impact Tarantino users - even if that's only 360 for now.

### PR DESCRIPTION
**Pull request by mpareja**
Closes #7.

## to-do

- [ ] - ~update documentation for how to specify source~
- [x] - remove TT_GIT_BASE and rely on the `com.tarantino.source` label for the full repo path
- [x] - update the template `docker-compose.yml` to demonstrate using jules/jackie

--------------------

This proposal would enable users to add a label to a docker-compose service. Take a look at the bottom of this example:

```
  tix:
    build: ./
    ports:
      - "8082:8081"
    environment:
      tix_graphite__host: 'graphite'
      tix_stats__host: 'graphite'
    volumes:
      - ./sql:/app/sql:ro

  deletemetix:
    build: ./
    ports:
      - "8082:8081"
    environment:
      tix_graphite__host: 'graphite'
      tix_stats__host: 'graphite'
    volumes:
      - ./sql:/app/sql:ro
    links:
      - tix
    labels:
      com.tarantino.source: '360incentives/delete-me-tix'
```

### CODE REVIEW

File: dc.sh [hunk 1]
```diff
@@ -79,5 +108,14 @@ generate_service_lists() {
 	local services_cache_file=$(get_workspace_cache_dir)/services
 	dc config --services > "$services_cache_file"
 
+	local sources_cache_file=$(get_workspace_cache_dir)/sources
+	parse_sources > $sources_cache_file
+
```
Comment (darrenmce) [inline dc.sh]: not sure if we need to cache this,  it's not used in autocomplete and I think the parseyaml is fast

**Comment by darrenmce** [comment]
I like the approach, the parse yaml I guess is a necessary evil. 

one note: ur example implies the TT_GIT_BASE would just be something like `git@github.com` and the source would just be the full path? is this intended or just example haste?

**Comment by mpareja** [comment]
Scratch that a bit. The value doesn't matter at all right now since the implementation is very specific. Just setting TT_GIT_BASE works.

**Comment by mpareja** [comment]
The point is, though. That in the future, it would be nice to support an npm-like approach where you can specify shortforms for GitHub (even bitbucket, etc.). Or some other custom git links.

**Comment by darrenmce** [comment]
yeah I'm fine either way, as long as we pick something that allows flexibility

**Comment by darrenmce** [comment]
(I say this so we can set the default git base/sample docker-compose to what we suggest)

**Comment by mpareja** [comment]
I added a to-do for updating samples. Are you cool if we start with one way of specifying source control and that is `<github org>/<github repo>` as is above? Should we wait until that parsing is implemented before merging this?

**Comment by darrenmce** [comment]
I think the only place we use it is `$TT_GIT_BASE/source` right? I mean as long as the base and the source make sense relative to each other it should work?

**Comment by darrenmce** [comment]
I'm almost tempted to just say put the full git path in the source and scrap the base for now, maybe add later as convenience

**Comment by darrenmce** [comment]
updated todo, though i still want to pull this down and try it at some point... and i will eventually (gotta find time)

**Comment by darrenmce** [comment]
just thought of a gotcha, but fixable gotcha

for when we are parsing the yaml, we should be running the parse against the `docker-compose config` output instead of the actual `.yml` file, since there is variable replacement that may be utilized in _both_ the `links` section and the `labels` sections (the get-links and the get-repos functions .. or what they are called now)

@mpareja

**Comment by mpareja** [comment]
Done!

**Comment by mpareja** [comment]
Very interesting! Want to create an issue?

I'm done with this PR. The documentation thing was me referring to adding things to the sample docker-compose file. Dude, it works!

File: dc.sh (hunk 1 above)
Comment (darrenmce) [inline dc.sh]: lets table this in favour of #12

**Comment by darrenmce** [comment]
might not get to this til tomorrow, fyi

**Comment by mpareja** [comment]
Yeah, if we end up with more usages, I'd consider adding a `-g` (`"declare -g -A TT_SOURCES=($(dc_get_sources))"`) and just using the `TT_SOURCES` variable as necessary.

**Pull request merged**